test/fork/migration-v2.js

```diff
@@ -584,6 +585,46 @@ describe('V2 upgrade', function () {
     this.claimsReward = newClaimsReward;
   });
 
+  it('Pool value check', async function () {
+    const poolValueAfter = await this.pool.getPoolValueInEth();
+    const poolValueDiff = poolValueAfter.sub(this.poolValueBefore);
+
+    const ethBalanceAfter = await ethers.provider.getBalance(this.pool.address);
+    const daiBalanceAfter = await this.dai.balanceOf(this.pool.address);
+    const stEthBalanceAfter = await this.stEth.balanceOf(this.pool.address);
+    const enzymeSharesBalanceAfter = await this.enzymeShares.balanceOf(this.pool.address);
+
+    console.log({
+      poolValueBefore: formatEther(this.poolValueBefore),
+      poolValueAfter: formatEther(poolValueAfter),
+      poolValueDiff: formatEther(poolValueDiff),
+      ethBalanceBefore: formatEther(this.ethBalanceBefore),
+      ethBalanceAfter: formatEther(ethBalanceAfter),
+      ethBalanceDiff: formatEther(ethBalanceAfter.sub(this.ethBalanceBefore)),
+      daiBalanceBefore: formatEther(this.daiBalanceBefore),
+      daiBalanceAfter: formatEther(daiBalanceAfter),
+      daiBalanceDiff: formatEther(daiBalanceAfter.sub(this.daiBalanceBefore)),
+      stEthBalanceBefore: formatEther(this.stEthBalanceBefore),
+      stEthBalanceAfter: formatEther(stEthBalanceAfter),
+      stEthBalanceDiff: formatEther(stEthBalanceAfter.sub(this.stEthBalanceBefore)),
+      enzymeSharesBalanceBefore: formatEther(this.enzymeSharesBalanceBefore),
+      enzymeSharesBalanceAfter: formatEther(enzymeSharesBalanceAfter),
+      enzymeSharesBalanceDiff: formatEther(enzymeSharesBalanceAfter.sub(this.enzymeSharesBalanceBefore)),
+    });
+
+    // Why 2 wei difference?
+    expect(poolValueDiff.abs(), 'Pool value in ETH should be the same').lessThanOrEqual(BigNumber.from(2));
+    expect(stEthBalanceAfter.sub(this.stEthBalanceBefore).abs(), 'stETH balance should be the same').lessThanOrEqual(
+      BigNumber.from(2),
```

Comment:
is there a 2 wei difference here?

if there is then it causes the overall pool value 2 wei diff

now as to why this happens, it could be some strangeness in the stETH internal math, i do remember they had off-by-1 wei errors when depositing ETH. 

you'd expect this to be exactly equal bc its a transfer but i wouldnt sweat too hard over it

Reply:
the stETH shares difference varies between 1 and 2 wei - unsure why tbh, but didn't have time to dig into their transfer function. could be related.

Reply:
yes that may be it, also some computation against the ETH backing amount, if you deposit 1 ETH it may give you 0.99999.. stETH.

we can live with this i think.